commonNativeUtils.gradle

```diff
@@ -98,27 +102,30 @@ def AddVSDllLinkOption(Project prj, String newOption) {
                     newOption
                 ]
             }
+            return [] // Always return at least an empty array, otherwise all the previous settings will be wiped out.
         }
     }
 }
 
 ////////////////////////////////////////////////////////////////
 def addDefaultCPPLibraryOptions(Project prj, CppLibrary lib, Boolean suppressConsoleDef=false) {
 
-    lib.targetMachines = [
-        prj.machines.windows.x86
-    ]
-
+    if (isWindows) {
+        lib.targetMachines = [prj.machines.windows.x86]
+    } else {
+        lib.targetMachines = [prj.machines.linux.x86_64]
+    }
```

Comment:
Here you should just set targetMachines to both contains both machines. The selection will be done by Gradle instead.

Reply:
I thought that I tried that and it didn't work, but I'll try again.